lib/Driver.cpp

```diff
@@ -63,25 +63,34 @@ std::vector<std::unique_ptr<TestResult>> Driver::Run() {
 
   for (auto &Test : Finder.findTests(Ctx)) {
     auto ObjectFiles = AllObjectFiles();
-    ExecutionResult ExecResult = Runner.runTest(Test.get(), ObjectFiles);
+
+    ExecutionResult ExecResult = Sandbox.run([&](ExecutionResult *SharedResult){
+      *SharedResult = Runner.runTest(Test.get(), ObjectFiles);
+    });
+
     auto BorrowedTest = Test.get();
     auto Result = make_unique<TestResult>(ExecResult, std::move(Test));
 
     for (auto Testee : Finder.findTestees(BorrowedTest, Ctx)) {
       auto ObjectFiles = AllButOne(Testee->getParent());
       for (auto &MutationPoint : Finder.findMutationPoints(MutationOperators, *Testee)) {
+        ExecutionResult R = Sandbox.run([&](ExecutionResult *SharedResult){
```

Comment:
> For each test/mutant run we 'boot' test framework from scratch (running all static constructors this case), and after the run we shut it down (running all destructors).
> I'm afraid we can't do it other way because of a state. 
> GoogleTest particularly does lots of things using static variables.

Yes, exactly. In mutants case (Ruby) performing the equivalent of static initializers (the gazillions of DSL call sides) can if shared as forced to be done "once" before the N forks significantly increase performance.

Hence I was just mentioning that once you observe "redunant work" being "re-done" per fork, you may be able to force this work to be inherited. Removing the time spend inside the fork significantly.